### ISSUE TITLE
Fix false negatives for dollar variables in *-notation

### ISSUE THREAD

**Issue by tao-cumplido** [stylelint/stylelint#5010]
> Clearly describe the bug

The `*-notation` rules added in 13.5.0 don't report values assigned to SCSS variables. They work fine when assigned to a CSS property directly.

> Which rule, if any, is the bug related to?

- `alpha-value-notation`
- `color-function-notation`
- `hue-degree-notation`

> What code is needed to reproduce the bug?

```scss
$a: rgba(0 0 0 / 0.5);
$c: rgb(0, 0, 0);
$h: hsl(0 0 0);
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "rules": {
    "alpha-value-notation": "percentage",
    "color-function-notation": "modern",
    "hue-degree-notation": "angle"
  }
}
```

> Which version of stylelint are you using?

`13.7.2`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

"CLI with `stylelint test.scss`"

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

"Yes, it's related to SCSS variables."

> What did you expect to happen?

each rule to report errors

> What actually happened (e.g. what warnings or errors did you get)?

no errors

**Comment by jeddy3** [comment]
Thanks for the report and for using the template.

I can reproduce this [using the demo](https://stylelint.io/demo#N4Igxg9gJgpiBcIAkBDeACATgcwEYoAoAGdEkgelIDoBWASgG4AdAOyTAx12IBpS+ijVkgAWGEQGcANsVKlGIHiAkBPFgBcUADwTKwEiYvAQWAMwCW2XcFbp0TEJgCuUmBIcYbLO3YcopAA4iKAC0AG7+TjAhLBCa6uYmHvYgATCYYDAaKNgwDjy2Pg6QUhCYIaZOLGAJJjFxKLUsyQ4AttDpzYqFviAiUSGw2Jgw0bHxiV0YfizYrg6FAL6siyCLQA).

I was over zealous in excluding non-standard syntax when I wrote these rules. All three rules return early if the _declaration_ that the _declaration-value_ belongs to is non-standard. In hindsight, I shouldn't have done this because the rules operate on the _declaration-value_, and whether the _declaration_ is standard or not is incidental.

We should exclude non-standard syntax in a consistent way, and I think we should do it on the most granular level. For example, if a rule operates on functions then only the `isStandardSyntaxFunction` is used, `isStandardSyntaxDeclaration` only for `declaration-*` rules and so on. I suggest we document this in developer guide.

I'll label this as bug (in line with our semantic versioning policy) and ready to implement.

@tao-cumplido Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide. You'll want to remove the `isStandardSyntaxDeclaration` conditional in each rule, and add a concise new `testRule` test for each rule, [like so](https://github.com/stylelint/stylelint/blob/8366b54f536b26c2a994865568336c577ea5625e/lib/rules/selector-combinator-space-before/__tests__/index.js#L489-L520).

**Comment by tao-cumplido** [comment]
Ok, I'll look into sometime this week.
